platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformBuilder.java

```diff
@@ -0,0 +1,375 @@
+     */
+    public PlatformBuilder withConfigPath(@NonNull final Path path) {
+        Objects.requireNonNull(path);
+        final Path absolutePath = getAbsolutePath(path);
+        if (!Files.exists(absolutePath)) {
+            throw new IllegalArgumentException("File " + absolutePath + " does not exist");
+        }
+        this.configPath = absolutePath;
+        return this;
+    }
+
+    /**
+     * Build a platform but do not start it.
+     *
+     * @return a new platform instance
+     */
+    public SwirldsPlatform build() {
+        return build(false);
+    }
+
+    /**
+     * Setup static utilities. If running multiple platforms in the same JVM, this should be called when constructing
+     * the first platform only.
+     *
+     * @param configuration the configuration for this node
+     * @return true if this is the first time this method has been called, false otherwise
+     */
+    private boolean setupStaticUtilities(@NonNull final Configuration configuration) {
+        if (staticSetupCompleted) {
+            // Only setup static utilities once
+            return false;
+        }
+        staticSetupCompleted = true;
+
+        ConfigurationHolder.getInstance().setConfiguration(configuration);
+
+        // Setup logging, using the configuration to tell us the location of the log4j2.xml
+        final PathsConfig pathsConfig = configuration.getConfigData(PathsConfig.class);
+        final Path log4jPath = pathsConfig.getLogPath();
+        try {
+            Log4jSetup.startLoggingFramework(log4jPath).await();
+        } catch (final InterruptedException e) {
+            CommonUtils.tellUserConsole("Interrupted while waiting for log4j to initialize");
+            Thread.currentThread().interrupt();
+        }
+
+        // Now that we have a logger, we can start using it for further messages
+        logger.info(STARTUP.getMarker(), "\n\n" + STARTUP_MESSAGE + "\n");
+        logger.debug(STARTUP.getMarker(), () -> new NodeStartPayload().toString());
+
+        BootstrapUtils.performHealthChecks(configPath, configuration);
+        writeSettingsUsed(configuration);
+
+        metricsProvider = new DefaultMetricsProvider(configuration);
+        globalMetrics = metricsProvider.createGlobalMetrics();
+        CryptoMetrics.registerMetrics(globalMetrics);
+
+        // Initialize the thread dump generator, if enabled via settings
+        startThreadDumpGenerator(configuration);
+
+        // Initialize JVMPauseDetectorThread, if enabled via settings
+        startJVMPauseDetectorThread(configuration);
+
+        return true;
+    }
+
+    /**
+     * Build the configuration for the node.
+     *
+     * @return the configuration
+     */
+    @NonNull
+    private Configuration buildConfiguration() {
+        if (configurationBuilder == null) {
+            configurationBuilder = ConfigurationBuilder.create();
+        }
+
+        if (!Files.exists(configPath)) {
```

Comment:
Can we move that checks to BootstrapUtils.setupConfigBuilder or even better to the config source that really uses the path?

Reply:
done